### PR TITLE
Integrate object builder into motoroid

### PR DESCRIPTION
**Pull request by mcjcloud**
Changes needed to integrate ObjectBuilder into Motoroid

## Changes

- Add build and upload methods to binding layer
- Replace object builder Go types with proto types
- Add QueryObject method to motor

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Review by prdlk — APPROVED** [review]

File: pkg/motor/query.go [hunk 1]
```diff
@@ -105,3 +106,11 @@ func (mtr *motorNodeImpl) QueryWhereIsByCreator(req mt.QueryWhereIsByCreatorRequ
 		WhereIs: ptrArr,
 	}, nil
 }
+
+func (mtr *motorNodeImpl) QueryObject(cid string) (map[string]interface{}, error) {
+	sh := shell.NewShell(mtr.Cosmos.GetIPFSApiAddress())
```
Comment (joshLong145) [inline pkg/motor/query.go]: there is a instance of shell already scoped within the `Motor`.

File: bind/motor-mobile/object_builder.go [hunk 2]
```diff
@@ -129,3 +130,54 @@ func RemoveObjectField(name, fieldName string) error {
...
+}
+
+func UploadObject(name string) ([]byte, error) {
+	if instance == nil {
+		return nil, errWalletNotExists
+	}
+
+	builder, ok := objectBuilders[name]
+	if !ok {
+		return nil, fmt.Errorf("no object builder with name '%s'", name)
```
Comment (joshLong145) [inline bind/motor-mobile/object_builder.go:160]: nit: perhaps we should move this a global var for object errors.

Comment (mcjcloud) [reply bind/motor-mobile/object_builder.go:160]: I'm not sure what you mean. A global object for storing errors that occur in motor?

File: pkg/motor/query.go (hunk 1 above)
Comment (mcjcloud) [reply pkg/motor/query.go]: I did not notice an instance in Motor but I added a reference to `motorNodeImpl`.

**Review by ntindle — APPROVED** [review]
Work with Ian to make sure this doesn't break speedway

File: bind/motor-mobile/object_builder.go (hunk 2 above)
Comment (joshLong145) [reply bind/motor-mobile/object_builder.go:160]: I mena more make it a global var in the bind layer. but we dont have this setup really yet. Ill check to see if there is an issue. resolving this comment.

**Review by joshLong145 — APPROVED** [review]

**Pull request merged**